OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -80,6 +80,10 @@ protected virtual bool CanAttack(Actor self, Target target)
 			if (!HasAnyValidWeapons(target))
 				return false;
 
+			var layer = self.Location.Layer;
+			if (layer == CustomMovementLayerType.Tunnel || layer == CustomMovementLayerType.Subterranean)
```

Comment:
That's what we do in https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Activities/Attack.cs#L105 (for AttackFrontal) and what I thought your old implementation was doing, but it still cancels the target.